ioctl/validator/validator.go

```diff
@@ -71,9 +71,13 @@ func ValidateStakeDuration(stakeDuration *big.Int) error {
 
 // ValidateCandidateName validates candidate name for native staking
 func ValidateCandidateName(candidateName string) error {
-	if len(candidateName) > 12 {
-		return ErrLongCandidateName
+	if len(s) == 0 || len(s) > 12 {
```

Comment:
current ValidateCandidateName() is also used for V1, should not change it

i think we'll need a new ValidateCandidateNameV2() for staking V2